nexusparser/tools/dataconverter/readers/base_reader.py

```diff
@@ -0,0 +1,26 @@
+"""The abstract class off of which to implement readers."""
+from abc import ABC, abstractmethod
+from typing import Tuple
+
+
+class BaseReader(ABC):
+    """
+    The abstract class off of which to implement readers.
+
+    The filename's prefix is the identifier. The '_reader.py' is snipped out.
+    For this BaseReader with filename base_reader.py the ID  becomes 'base'
+
+    For future reference:
+    - Support links by setting the path in the template with the following object
+       object = {"link": "/path/to/source/data"}
+    """
+
+    # pylint: disable=too-few-public-methods
+
+    @abstractmethod
+    def read(self, template: dict = None, file_paths: Tuple[str] = None) -> dict:
```

Comment:
We do requilre multiple input files being passed to the reader. Sometimes data comes from different detectors on the same instrument and results in multiple data files. I decided to explicitly list it like this to avoid any confusion from the community developing the readers. The question will then come up on how to handle multiple file inputs in the reader implementation. So if you're okay with it, I'll leave this explicitly shown rather than hidden behind *args. I hope I got your use of *args right. :D